src/code_080023A4.c

```diff
@@ -161,3 +277,4739 @@ void sub_080027C8(void)
+{
+    if (gUnk_0203AD2C < 200) {
+        *arg1 = gUnk_082D8CB8[arg0];
+        *arg2 = gUnk_082D8CC0[arg0];
+
+        arg3[0] = gUnk_082D8CD0[arg0][0];
+        arg3[1] = gUnk_082D8CD0[arg0][1];
+    }
+    else if (gUnk_0203AD2C < 300) {
+        *arg1 = gUnk_082D8CF0[arg0];
+        *arg2 = gUnk_082D8CF8[arg0];
+
+        arg3[0] = gUnk_082D8D08[arg0][0];
+        arg3[1] = gUnk_082D8D08[arg0][1];
+    }
+    else {
+        *arg1 = gUnk_082D8D28[arg0];
+        *arg2 = gUnk_082D8D30[arg0];
+
+        arg3[0] = gUnk_082D8D40[arg0][0];
+        arg3[1] = gUnk_082D8D40[arg0][1];
+    }
+}
+
+void sub_08002D40(u16 arg0, u8 *arg1, u8 *arg2)
+{
+    *arg1 = gUnk_0893CB44[arg0][0];
+    *arg2 = gUnk_0893CB44[arg0][1];
+}
+
+void sub_08002D60(u32 arg0, struct LevelInfo *levelInfo)
+{
+    struct ObjectBase *var0 = (void *) (gKirbys + levelInfo->unk660);
+
+    levelInfo->unk1C = -(levelInfo->unkC  + 0x7800) + var0->x;
+    levelInfo->unk20 = -(levelInfo->unk10 + 0x5000) + var0->y;
+
+    sub_08002DA0(arg0, levelInfo);
+}
+
+void sub_08002DA0(u32 arg0, struct LevelInfo *levelInfo)
+{
+    s32 var0 = levelInfo->unkC  + levelInfo->unk1C;
+    s32 var1 = levelInfo->unk10 + levelInfo->unk20;
+
+    if (var0 < levelInfo->unk6C)
+        levelInfo->unk1C += levelInfo->unk6C - var0;
+
+    var0 += 0xF000;
+
+    if (var0 > levelInfo->unk74)
+        levelInfo->unk1C -= var0 - levelInfo->unk74;
+
+    if (var1 < levelInfo->unk70)
+        levelInfo->unk20 += levelInfo->unk70 - var1;
+
+    var1 += 0xA000;
+
+    if (var1 > levelInfo->unk78)
+        levelInfo->unk20 -= var1 - levelInfo->unk78;
+}
+
+static void nullsub_10(void) {}
+
+void sub_08002DFC(u8 arg0, struct LevelInfo *arg1)
+{
+    struct LevelInfo *var0 = gCurLevelInfo + arg1->unk661;
+
+    if (arg1->currentRoom == var0->currentRoom) {
+        arg1->unkC  = var0->unkC;
+        arg1->unk10 = var0->unk10;
+        arg1->unk1C = 0;
+        arg1->unk20 = 0;
+    }
+}
+
+void sub_08002E3C(void)
+{
+    gUnk_02023350 = 0;
+}
+
+void sub_08002E48(struct Unk_08002E48_0 *arg0)
+{
+    u32 var0;
+    struct Unk_08002E48_1 *var1;
+    u8 p;
+
+    var0 = (arg0->unk12 & 0x10) != 0 ?
+        0x2000000 + arg0->unk6 * 4 :
+        0x3000000 + arg0->unk6;
+
+    var1 = (void *) var0;
+
+    for (p = 0; p < 4; p++)
+        if (gCurLevelInfo[p].currentRoom != 0xFFFF)
+            sub_08003028(p, p == var1->unk0);
+
+    gUnk_03002E60 = var1->unk24;
+    gUnk_02023354 = 0;
+}
+
+void sub_08002EC4(u8 playerId, bool32 arg1)
+{
+    switch (gCurLevelInfo[playerId].currentRoom) {
+        case 918:
+            if (!arg1) break;
+
+            gCurLevelInfo[playerId].unk8 = 7;
+
+            gDispCnt &= ~7;
+            gDispCnt |= DISPCNT_MODE_1;
+
+            gBgCntRegs[2] = (gBgCntRegs[2] & ~BGCNT_AFF1024x1024) | BGCNT_AFF256x256;
+            gBgCntRegs[2] = (gBgCntRegs[2] & ~BGCNT_PRIORITY(3)) | BGCNT_PRIORITY(2);
+
+            gDispCnt &= ~DISPCNT_BG3_ON;
```

Comment:
I think you either need to define mask macros or just leave them as literals. The current code is not showing the fact that all the related bits are cleared. 
```suggestion
            gBgCntRegs[2] = (gBgCntRegs[2] & ~0xC000) | BGCNT_AFF256x256;
            gBgCntRegs[2] = (gBgCntRegs[2] & ~3) | BGCNT_PRIORITY(2);

            gDispCnt &= ~DISPCNT_BG3_ON;
```